cusub_cortex/state_machine/src/tasks/start_gate_det.py

```diff
@@ -0,0 +1,685 @@
+                remapping={'timeout_obj':'timeout_obj', 'outcome':'outcome'})
+            # FILL IN REST OF OBJECTIVES
+
+
+class Approach_1_Pole(Objective):
+    outcomes = ['found_2nd_pole', 'timed_out', 'lost_object']
+
+    target_class_id = "start_gate_pole"
+
+    def __init__(self, task_name, listener):
+        name = task_name + "/Approach_1_Pole"
+        super(Approach_1_Pole, self).__init__(self.outcomes, name)
+        self.listener = listener
+        self.yaw_client = PIDClient(name, "yaw")
+        self.drive_client = PIDClient(name, "drive") #, "cusub_common/motor_controllers/mag_pid/")
+        self.depth_client = PIDClient(name, "depth")
+
+        seconds = rospy.get_param("tasks/start_gate/seconds_in_postion", 2)
+        self.rate = rospy.get_param("tasks/start_gate/new_dv_check_rate", 30)
+        self.count_target = seconds * self.rate
+        self.count = 0
+
+        self.mag_target = rospy.get_param("tasks/start_gate/mag_target")
+        self.elev_target = rospy.get_param("tasks/start_gate/elev_target")
+        self.elev_thresh = rospy.get_param("tasks/start_gate/elev_thresh")
+
+    def execute(self, userdata):
+        self.cuprint("executing")
+        self.toggle_waypoint_control(True)
+
+        # Find start_gate pole dobject number and check for errors
+        dobj_nums = self.listener.query_class(self.target_class_id)
+        if len(dobj_nums) > 0: # Check if more than 1 instance of target class
+            self.cuprint(str(len(dobj_nums)) + " " + self.target_class_id + " classe(s) detected!", warn=True)
+            self.cuprint("selecting the first", warn=True)
+        elif not dobj_nums: # Check if target class is not present (shouldn't be possible)
+            self.cuprint("somehow no " + self.target_class_id + " classes found?", warn=True)
+            return "lost_object"
+        # DOBJECT NUMBER
+        self.cuprint(str(dobj_nums), warn=True)
+        dobj_num = dobj_nums[0]
+        self.cuprint("located " + self.target_class_id + "'s dobject num: " + str(dobj_num))
+
+        # TODO start a watch dog timer on detections
+
+        # Enable the PID loops
+        self.yaw_client.enable()
+        self.drive_client.enable()
+        self.depth_client.enable()
+        self.drive_client.set_setpoint(self.mag_target)
+        self.depth_client.set_setpoint(-1.5)
+        self.in_position_state = None
+
+        self.cuprint("servoing")
+        r = rospy.Rate(self.rate)
+        while not rospy.is_shutdown():
+            if self.listener.check_new_dv(dobj_num):
+                [az, el, mag] = self.listener.get_avg_bearing(dobj_num, num_dv=5)
+                mag = np.sqrt(mag)
+
+                self.yaw_client.set_setpoint(az, loop=False)
+                if abs(self.mag_target - mag) < 0.5*self.mag_target:
+                    self.cuprint("[MAGNITUDE: " + str(mag) + "] - " + "[AZ: " + str(az) + "] - " +  "[CHECK POSITION: " + str(self.count) + " / " + str(self.count_target) + "]")
```

Comment:
there's a new version of cuprint that allows you to write just on one line. It does a reverse newline so you can overwrite what was just printed and stay on the same line

Reply:
see latest version of droppers when it's printing status